OpenNefia.Content/UI/Hud/Widgets/UiHudMinimap.cs

```diff
@@ -9,26 +9,38 @@
 using System.Text;
 using System.Threading.Tasks;
 using OpenNefia.Core.Maths;
+using OpenNefia.Core.GameObjects;
+using OpenNefia.Core.Game;
 
-namespace OpenNefia.Content.UI.Hud
+namespace OpenNefia.Content.Hud
 {
-    public class Minimap : BaseDrawable
+    public class UiHudMinimap : BaseHudWidget
     {
         [Dependency] private readonly ITileDefinitionManager _tileDefManager = default!;
+        [Dependency] private readonly IEntityManager _entMan = default!;
+        [Dependency] private readonly IMapManager _mapManager = default!;
+
         private Vector2i MapSize;
         private MapCoordinates? PlayerCoords = default!;
-        private TileAtlasBatch Batch;
-        private IAssetInstance PlayerIcon;
+        private TileAtlasBatch Batch = default!;
+        private IAssetInstance PlayerIcon = default!;
 
         private readonly Color WallColor = new(90, 90, 90);
 
-        public Minimap()
+        public override void Initialize()
         {
-            IoCManager.InjectDependencies(this);
+            base.Initialize();
             Batch = new TileAtlasBatch(AtlasNames.Tile);
             PlayerIcon = Assets.Get(Protos.Asset.MinimapMarkerPlayer);
         }
 
+        public override void UpdateWidget()
+        {
+            base.UpdateWidget();
+            if (_entMan.TryGetComponent<SpatialComponent>(GameSession.Player, out var spatial))
+                Refresh(_mapManager.ActiveMap?.TileMemory!, spatial.MapPosition);
```

Comment:
Assume that `ActiveMap` is not null here.

Reply:
removing either the ? or ! makes it not compile

Reply:
I was meaning more of doing `_mapManager.ActiveMap!.TileMemory` instead of two null operators.

Reply:
ah didn't know !s could be placed in that position at all, changed in 39dc769